### DISCUSSION TITLE
The Cobbler server has been configured and the client cannot boot

### DISCUSSION THREAD

**Discussion by ou332697296** [cobbler/cobbler#2742] [Version specific questions]
Hello everyone, I have the server has been configured ok, but the client cannot properly boot prompt PXE-E23, the following is the error code :
server IP address is 192.168.0.101
NBP filename is grub/grubx64.efi
NBP filename is 0 Bytes
PXE-E23:  Client received TFTP error from server.

os:centos8
version:3.2.0

This is my Settings file:
[settings.txt](https://github.com/cobbler/cobbler/files/6930550/settings.txt)
Here is my DHCP configuration file
[dhcp.template.txt](https://github.com/cobbler/cobbler/files/6930552/dhcp.template.txt)

I don't know how to handle this error, anyone?  Thank you very much.

**Comment by ericchambers1940** [comment]
I’ll give this a shot.
First:

- Are you attempting a UEFI install? Do legacy bios pxe boots work?
- Have you ensured all of your basic Cobbler related services are running without errors (tftpd, cobblerd, xinetd, httpd)?
- Have you sifted through /var/log/messages for any more verbose errors pertaining to tftp?
- What does /var/log/cobbler/cobbler.log tell you when you attempt to boot?

Preliminary Google searches indicate pxe error code e23 accompanied with a “0 byte” npb file indicate missing boot programs. See here for an example (it’s Windows related but should still apply since pxe is an industry wide standard):
https://goddamnpc.com/how-to-fix-nbp-filesize-is-0-bytes-when-pxe-booting-pxe-e23-error/

I believe you’ll need to run a script called mkgrub.sh (someone correct me if I’m wrong). It obtains missing boot loaders (or you might have to manually copy one from a working system). I don’t remember where it’s located on the cobbler installation but you can run “find / -name “*mkgrub.sh*” and it should show up.

afterwards, I think you need to run cobbler sync.

-Eric
